### PR TITLE
Random Battle: Add possibility of Z-Conversion

### PR DESCRIPTION
**Pull request by ascriptmaster**

### CODE REVIEW

**Comment by ascriptmaster** [comment]
Right now, with the current setup of rejecting Conversion with Tri-Attack, I ran some sims and it seems the chance of getting the Z-Conversion set is about 1/8. Is this acceptable?

I tried rejecting Tri-Attack with Conversion, but that gave it over a 50% chance of getting a Z-Conversion set , and that set is always the same.

Yes, Dark Pulse > Shadow Ball slightly outside of Z-Conversion sets because of its ability to hit Normals, but I think this will be fine regardless.

File: data/scripts.js [hunk 1]
```diff
@@ -1860,7 +1864,7 @@ exports.BattleScripts = {
 					break;
 				}
 			}
-			if (moves.length === 4 && !counter.stab && !hasMove['metalburst'] && (counter['physicalpool'] || counter['specialpool'])) {
+			if (moves.length === 4 && !counter.stab && !hasMove['conversion'] && !hasMove['metalburst'] && (counter['physicalpool'] || counter['specialpool'])) {
```
Comment (TheImmortal) [inline data/scripts.js]: Why this?

Comment (TheImmortal) [inline data/scripts.js]: Never mind, I know why. But there must be a better way.

Comment (ascriptmaster) [inline data/scripts.js]: Maybe in `queryMoves` do something like this:

`if (move.priority === 0 && (hasAbility['Protean'] || moves.includes('conversion')) && !(moveid in NoStab)) counter['stab']++;`

**Comment by TheImmortal** [comment]
You need to reject Conversion with Trick, or vice versa. Also, I say remove Agility.

**Comment by ascriptmaster** [comment]
By specifying Conversion as a `MixedSetup` move in the `queryMoves` section, Trick auto-rejects with it because Porygon-Z will have `counter.setupType === 'Mixed'` 

				case 'switcheroo': case 'trick':
					if (counter.Physical + counter.Special < 3 || counter.setupType) rejected = true;

**Comment by ascriptmaster** [comment]
Removed Agility, updated Conversion to set the STAB counter in `queryMoves`.

**Pull request merged**